test/SlowTests/Server/Documents/ETL/EtlTestBase.cs

```diff
@@ -78,5 +100,94 @@ protected ManualResetEventSlim WaitForEtl(DocumentStore store, Func<string, EtlP
+                    taskCompletionSource.SetException(e);
+                }
+            }
+
+            database.EtlLoader.BatchCompleted += EtlLoaderOnBatchCompleted;
+            var whenAny = await Task.WhenAny(taskCompletionSource.Task, Task.Delay(timeout));
+            database.EtlLoader.BatchCompleted -= EtlLoaderOnBatchCompleted;
+
+            if(whenAny != taskCompletionSource.Task)
+                throw new TimeoutException($"Etl predicate timeout - {timeout}");
+
+            return await taskCompletionSource.Task;
+        }
+
+        private async Task<string[]> GetEtlErrorNotifications(DocumentStore src)
+        {
+            var databaseInstanceFor = await GetDocumentDatabaseInstanceFor(src);
+            using (databaseInstanceFor.NotificationCenter.GetStored(out IEnumerable<NotificationTableValue> storedNotifications, postponed: false))
+            {
+                var notifications = storedNotifications
+                    .Select(n => n.Json)
+                    .Where(n => n.TryGet("AlertType", out string type) && type.StartsWith("Etl_"))
+                    .Where(n => n.TryGet("Details", out BlittableJsonReaderObject _))
+                    .Select(n =>
+                    {
+                        n.TryGet("Details", out BlittableJsonReaderObject details);
+                        return details.ToString();
+                    }).ToArray();
+                return notifications;
+            }
+        }
+        
+        protected IAsyncDisposable OpenEtlOffArea(IDocumentStore store, long etlTaskId, bool cleanTombstones = false)
+        {
+            store.Maintenance.Send(new ToggleOngoingTaskStateOperation(etlTaskId, OngoingTaskType.RavenEtl, true));
+            return new DisposableAsyncAction(async () =>
+            {
+                if (cleanTombstones)
+                {
+                    var srcDatabase = await GetDatabase(store.Database);
+                    await srcDatabase.TombstoneCleaner.ExecuteCleanup();    
+                } 
+                
+                store.Maintenance.Send(new ToggleOngoingTaskStateOperation(etlTaskId, OngoingTaskType.RavenEtl, false));
+            });
+        }
+
+        public override void Dispose()
+        {
+            try
+            {
+                if (_src != null)
+                {
+                    var b = Context.TestException;
+                    var notifications = GetEtlErrorNotifications(_src).Result;
+
+                    var a = Context.TestOutput?.ToString();
+                    
+                    if (notifications.Any() && Context.TestException != null && Context.TestOutput != null)
+                    {
+                        string message = string.Join(",\n", notifications);
+                        Context.TestOutput.WriteLine(message);
+                    }
+                }
+            }
+            catch
+            {
+                // ignored
+            }
+
+            base.Dispose();
```

Comment:
I would put it in `finally`. Effectively it's the same thing in current impl but to underline that his needs to be _always_ called